api_gen.py

```diff
@@ -184,9 +199,9 @@ def wrapif(condition, code):
     def write_raw_sig(self):
         """ Write out "cdef extern"-style definition for an HDF5 function """
 
-        raw_sig = "{0.code} {0.fname}({0.sig}) except *\n".format(self.line)
+        raw_sig = "{0.code} {0.fname}({0.sig}) nogil except *\n".format(self.line)
```

Comment:
Is this accidentally leftover? I'd rather any changes that affect what the code does were separate from the more mechanical updating of the code.

Reply:
Hi Thomas,
No, it is not "accidentally" left-over. I can ensure you it causes no harm as the libhdf5 does not manipulate python object (except within callback but this should be handled by the callback).
Nogil function can be called with or without the gil; but only nogil functions can be called with the gil released.

I consider in a second PR to annotate every single function of the libhdf5 which does not use any (python-) callback to call them within a nogil section. Lots of manual reading, little code.

The last part and probably the most complicated is to duplicate all functions in _conv with a nogil variant to be used as callback in nogil sections, probably re-acquireing the gil.

Reply:
OK, sorry. I thought you had intended to drop all the nogil bits from this and just make the syntax changes.

I don't feel that I understand well enough how `nogil` interacts with Cython code, so I'll let someone else review that.

Reply:
I'd be happy to merge this quickly except for this change, which would need me to spend time reading up on Cython. Is it necessary for the rest of the PR? If not, can I suggest that we revert this bit and do nogil work in a separate PR from the syntax changes.

Reply:
As you wish